src/gcp/auth.ts

```diff
@@ -36,3 +86,49 @@ export async function updateAuthDomains(project: string, authDomains: string[]):
   );
   return res.body.authorizedDomains;
 }
+
+/**
+ * getAuthUser
+ * @param project project identifier.
+ * @param email the users email to lookup.
+ * @param phone the users phone number to lookup.
+ * @param uid the users id to lookup.
+ * @param limit the amount of results to return.
+ * @param offset the amount to page through the results
+ * @return an array of user info
+ */
+export async function getAuthUser(
+  project: string,
+  email?: string,
+  phone?: string,
+  uid?: string,
+  limit = 1000,
```

Comment:
This seems wonky... I'd expect getAuthUser to return a single UserInfo, not an array.
I'd recommend renaming this to `searchAuthUser`. 

Alternatively, you could:
- get rid of the limit and offset arguments, and switch the return type to UserInfo
- hard code the limit to 1 & unpack the results within this function
- return an error if there are no matching users

Reply:
I was hoping to use this for 2 seperate tools. The first is to get one and only one user, the second is to list all users or search for a particular user.

Reply:
I'd probably have separate `listUsers({project,limit?})` and `findUser({project,email?,uid?,phone?})` commands, it's a little weird to do both in one.

Reply:
Ok, I can do that.